lib/MatterbridgeManager.php

```diff
@@ -642,6 +642,9 @@ public function getCurrentVersionFromBinary(): ?string {
 			return null;
 		}
 
+		$cmd = sprintf('chmod u+x %s', escapeshellarg($binaryPath));
+		@exec($cmd, $output, $returnCode);
```

Comment:
I would like to not do this. This is job of the sysadmin and nothing we should fix in the software.

Reply:
What do you mean? In the current state, it does not work because extracted binaries don't have execution permissions (at least on my system).

Are you suggesting we tell the admin to `chmod`?

Reply:
For me the items from https://apps.nextcloud.com/apps/talk_matterbridge are executable when extracting the package.

Reply:
But yeah, the admin should decide on this and fix it. I'm too scared to have code in place that automatically makes a given config string executable and executes it.

Reply:
> For me the items from https://apps.nextcloud.com/apps/talk_matterbridge are executable when extracting the package.

I guess it depends on Php config. Mine is default from Ubuntu and binaries are 644 when extracted.

> the admin should decide on this and fix it

Ok then should we check permissions and display the suggested fix only if necessary?

Reply:
yeah, just show something like "Binary is not executable, please make sure the file is owned by the correct user and can be executed." Maybe you can copy the logic from https://github.com/nextcloud/server/blob/cb057829f72c70e819f456edfadbb29d72dba832/console.php#L67-L76 to show the needed user